x-pack/plugins/security_solution/cypress/e2e/detection_rules/bulk_edit_rules_actions.cy.ts

```diff
@@ -0,0 +1,186 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ROLES } from '../../../common/test';
+
+import {
+  RULES_BULK_EDIT_ACTIONS_INFO,
+  RULES_BULK_EDIT_ACTIONS_WARNING,
+  ADD_RULE_ACTIONS_MENU_ITEM,
+} from '../../screens/rules_bulk_edit';
+
+import { cleanKibana, deleteAlertsAndRules, deleteConnectors } from '../../tasks/common';
+import {
+  addSlackRuleAction,
+  assertSlackRuleAction,
+  addEmailConnectorAndRuleAction,
+  assertEmailRuleAction,
+} from '../../tasks/common/rule_actions';
+import {
+  waitForRulesTableToBeLoaded,
+  selectNumberOfRules,
+  loadPrebuiltDetectionRulesFromHeaderBtn,
+  goToEditRuleActionsSettings,
+} from '../../tasks/alerts_detection_rules';
+import {
+  waitForBulkEditActionToFinish,
+  submitBulkEditForm,
+  checkOverwriteRuleActionsCheckbox,
+  openBulkEditRuleActionsForm,
+  pickActionFrequency,
+  openBulkActionsMenu,
+} from '../../tasks/rules_bulk_edit';
+import { assertSelectedActionFrequency } from '../../tasks/edit_rule';
+import { login, visitWithoutDateRange } from '../../tasks/login';
+import { esArchiverResetKibana } from '../../tasks/es_archiver';
+
+import { SECURITY_DETECTIONS_RULES_URL } from '../../urls/navigation';
+
+import {
+  createMachineLearningRule,
+  createCustomIndicatorRule,
+  createEventCorrelationRule,
+  createThresholdRule,
+  createNewTermsRule,
+  createSavedQueryRule,
+  createCustomRuleEnabled,
+} from '../../tasks/api_calls/rules';
+import { createSlackConnector } from '../../tasks/api_calls/connectors';
+
+import {
+  getEqlRule,
+  getNewThreatIndicatorRule,
+  getNewRule,
+  getNewThresholdRule,
+  getMachineLearningRule,
+  getNewTermsRule,
+} from '../../objects/rule';
+
+const expectedNumberOfCustomRulesToBeEdited = 7;
+// 7 custom rules of different types + 3 prebuilt.
+// number of selected rules doesn't matter, we only want to make sure they will be edited an no modal window displayed as for other actions
+const expectedNumberOfRulesToBeEdited = expectedNumberOfCustomRulesToBeEdited + 3;
+
+const expectedSlackMessage = 'Slack action test message';
+
+describe('Detection rules, bulk edit of rule actions', () => {
+  before(() => {
+    cleanKibana();
+    login();
+  });
+
+  beforeEach(() => {
+    deleteAlertsAndRules();
+    deleteConnectors();
+    esArchiverResetKibana();
+
+    createCustomRuleEnabled(getNewRule(), '1');
+    createEventCorrelationRule(getEqlRule(), '2');
+    createMachineLearningRule(getMachineLearningRule(), '3');
+    createCustomIndicatorRule(getNewThreatIndicatorRule(), '4');
+    createThresholdRule(getNewThresholdRule(), '5');
+    createNewTermsRule(getNewTermsRule(), '6');
+    createSavedQueryRule({ ...getNewRule(), savedId: 'mocked' }, '7');
+
+    createSlackConnector();
+  });
+
+  context('Restricted action privileges', () => {
+    it('User has no actions privileges', () => {
```

Comment:
Let's rename this test to something more descriptive, like "User with no privileges can't add rule actions".